### ISSUE TITLE
Android Permission "Allow to make and manage phone calls" due to call to SystemInfo.deviceUniqueIdentifier

### ISSUE THREAD

**Issue by turkenh** [googleanalytics/google-analytics-plugin-for-unity#129]
Using the plugin results for addition of the permission "Allow to make and manage phone calls" which is scary. A solution is needed. 
It seems that this is a result of the call to SystemInfo.deviceUniqueIdentifier in the following line:

https://github.com/googleanalytics/google-analytics-plugin-for-unity/blob/master/source/Plugins/GoogleAnalyticsV3/GoogleAnalyticsMPV3.cs#L62

**Comment by acrolink** [comment]
Looks like a bug to me that the Android team must address. I am getting the same scary warning **"Allow to make and manage phone calls"** when I need to access `TelephonyManager` to read the IMEI.

**Comment by Zamaroht** [comment]
Yes, I think this should be addressed. There was another issue about the same problem:
https://github.com/googleanalytics/google-analytics-plugin-for-unity/issues/15

There are other ways to get an unique identifier that don't require the READ_PHONE_STATE permission and could be used instead.

**Comment by turkenh** [comment]
Any update here???

**Comment by miguel12345** [comment]
Hi,

So, I faced the same issue, and at first hand, I also thought this was a problem with Google Analytics. But if you look carefully, you can see that the entire `GoogleAnalyticsMPV3.cs` is essentially empty after macro processing due to the check made at https://github.com/googleanalytics/google-analytics-plugin-for-unity/blob/master/source/Plugins/GoogleAnalyticsV3/GoogleAnalyticsMPV3.cs#L30

So, in short, Google Analytics plugin SDK **does not use `SystemInfo.deviceUniqueIdentifier` for android builds**. You probably have an external party plugin that is using it. In our case, we had an external DLL that referenced `SystemInfo.deviceUniqueIdentifier` inside. After removing it, that permission was no longer present in the final manifest.

**Comment by AldoMaticDon** [comment]
I updated the biz snap app. And yes what the devil could they be up to.  Will my phone be used as a Marketing device.   ?  I want to go back to the version that came with my phone.   
Can a person clone their OS on a phone like we do when we clone a Hard drive with acronis.  ?  
They are putting organizations in a bind because this Biz Snap is part of workflows.   Its a stick in the spokes.     I would like to know how to

**Comment by megumin-firemage** [comment]
This is not code related but does pertain to the issue at hand:

This issue popped up when I removed egregious permissions from various apps on my tablet device. This tablet is not SIM equipped and only connects via WIFI, making traditional telephony unnecessary as it uses internet protocols. When attempting to use my camera just now, I was greeted with a message that I would have to enable my camera to Make and Manage phone calls... now wait one moment... Phone calls on a device that is not intended to be used as a phone is one thing, but a camera making and managing phone calls? That is a problem for me and for anyone who wonders just how invasive Google is getting with our devices. Google Play Services enables every basic permission by default. I have removed all permissions for that application and my Gmail application now greets me with a message that the app will not work properly unless I re-engage permissions for Google Play Services to access my contacts and my Phone... I have not had any issues using Gmail since revoking these permissions so I will continue use and simply ignore these messages. I can understand what Gmail would use contacts for, though revoking that permission has not impeded my ability to use the app. What Gmail would need access to my phone for is, again, beyond me.
